This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20L=C3=B6ffler?= <st.loeffler@gmail.com>
Date: Mon, 6 Nov 2017 06:59:37 +0100
Subject: [PATCH 1/1] Use a custom font directory on Windows

Uses the share/fonts directory alongside the application/poppler library

diff --git a/poppler/GlobalParams.cc b/poppler/GlobalParams.cc
index 1111111..2222222 100644
--- a/poppler/GlobalParams.cc
+++ b/poppler/GlobalParams.cc
@@ -150,6 +150,36 @@ static const char *get_poppler_datadir(void)
 #    undef POPPLER_DATADIR
 #    define POPPLER_DATADIR get_poppler_datadir()
 
+static const char *
+get_poppler_fontsdir (void)
+{
+  static char retval[MAX_PATH];
+  static int beenhere = 0;
+
+  unsigned char *p;
+
+  if (beenhere)
+    return retval;
+
+  if (!GetModuleFileName (hmodule, (CHAR *) retval, sizeof(retval) - 20))
+    return POPPLER_DATADIR;
+
+  p = _mbsrchr ((unsigned char *) retval, '\\');
+  *p = '\0';
+  p = _mbsrchr ((unsigned char *) retval, '\\');
+  if (p) {
+    if (stricmp ((const char *) (p+1), "bin") == 0)
+      *p = '\0';
+  }
+  strcat (retval, "\\share\\fonts");
+
+  beenhere = 1;
+
+  return retval;
+}
+#undef POPPLER_FONTSDIR
+#define POPPLER_FONTSDIR get_poppler_fontsdir ()
+
 #endif
 
 //------------------------------------------------------------------------
@@ -845,7 +875,7 @@ GooString *GlobalParams::findFontFile(const GooString *fontName)
 {
     GooString *path = nullptr;
 
-    setupBaseFonts(nullptr);
+    setupBaseFonts(POPPLER_FONTSDIR);
     globalParamsLocker();
     const auto fontFile = fontFiles.find(fontName->toStr());
     if (fontFile != fontFiles.end()) {
diff --git a/poppler/GlobalParamsWin.cc b/poppler/GlobalParamsWin.cc
index 1111111..2222222 100644
--- a/poppler/GlobalParamsWin.cc
+++ b/poppler/GlobalParamsWin.cc
@@ -504,7 +504,7 @@ GooString *GlobalParams::findSystemFontFile(const GfxFont *font, SysFontType *ty
     if (!fontName)
         return nullptr;
     std::unique_lock<std::recursive_mutex> locker(mutex);
-    setupBaseFonts(nullptr);
+    setupBaseFonts(POPPLER_FONTSDIR);
 
     // TODO: base14Name should be changed?
     // In the system using FontConfig, findSystemFontFile() uses
